core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -460,7 +460,9 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
         |                              on one of the worker machines inside the cluster ("cluster")
         |                              (Default: client).
         |  --class CLASS_NAME          Your application's main class (for Java / Scala apps).
-        |  --name NAME                 A name of your application.
+        |  --name NAME                 A name of your application. In yarn-cluster mode the name
```

Comment:
@vanzin Okay. The problem is with `yarn-client` mode. When you have an application name with `SparkContext.setAppName` and you use the `spark-submit` shell script with `--name` as well, `SparkContext.setAppName` will be used. `--name` should have precedence then. Would it be okay, if we would change the code to use `--name` always - if supplied - and to encourage users at `SparkContext.setAppName` to use `--name` if possible?